src/Task.js

```diff
@@ -51,14 +51,39 @@ const doM = steps => Task((err, res) => {
   step().fork(err, res)
 })
 
+// foldM :: (a -> b -> Task e a) -> a -> [b] -> Task e a
+const foldM = (f, init) => (ms) =>
+  Task((rej, res) => {
+    let cancelled = undefined
+    let result = init
+    const _ = (i) => {
+      if (i >= ms.length) {
+        res(result)
+      } else if (cancelled === undefined) {
+        f(result, ms[i]).fork(rej, (x) => {
+          result = x
+          setTimeout(_, 21, i + 1)
```

Comment:
Might be good idea to parameterise the delay as a third argument to foldM: `foldM(f, init, {delay:21})` ?

Reply:
Good idea.  Will add that.

Reply:
Done